pallets/subtensor/src/block_step.rs

```diff
@@ -4,6 +4,13 @@ use sp_std::vec::Vec;
 use substrate_fixed::types::I110F18;
 use substrate_fixed::types::I64F64;
 
+struct SubnetInfo {
```

Comment:
I wonder if we could make the code DRY by reusing the `SubnetInfo` struct from `subnet_info.rs`.

Reply:
We probably can, but am not 100% sure. We could use `DynamicPoolInfo` instead of `SubnetInfo` in `subnet_info.rs` because that one doesn't have `is_dynamic`, `price`, or `tao_staked`. Though I don't think `is_dynamic` field will be relevant to `DynamicPoolInfo` because we are only supposed to use it for dynamic networks. 

As a side note, collecting all data structures in some kind of types.rs module could be helpful.